deliver/lib/deliver/submit_for_review.rb

```diff
@@ -1,132 +1,169 @@
 require_relative 'module'
 
+require 'fastlane_core/build_watcher'
+require 'fastlane_core/ipa_file_analyser'
+require 'fastlane_core/pkg_file_analyser'
+
 module Deliver
   class SubmitForReview
     def submit!(options)
-      app = options[:app]
-      select_build(options)
-
-      UI.message("Submitting the app for review...")
-      submission = app.create_submission(platform: options[:platform])
-
-      # Set app submission information
-      # Default Values
-      submission.content_rights_contains_third_party_content = false
-      submission.content_rights_has_rights = true
-      submission.add_id_info_uses_idfa = false
-
-      # User Values
-      if options[:submission_information]
-        options[:submission_information].each do |key, value|
-          UI.message("Setting '#{key}' to '#{value}'...")
-          submission.send("#{key}=", value)
-        end
+      legacy_app = options[:app]
+      app_id = legacy_app.apple_id
+      app = Spaceship::ConnectAPI::App.get(app_id: app_id)
+
+      platform = Spaceship::ConnectAPI::Platform.map(options[:platform])
+      version = app.get_edit_app_store_version(platform: platform)
+
+      unless version
+        UI.user_error!("Cannot submit for review - could not find an editable version for '#{platform}'")
+        return
       end
 
-      # Finalize app submission
-      submission.complete!
+      build = select_build(options, app, version, platform)
+
+      update_export_compliance(options, app, build)
+      update_idfa(options, app, version)
+      update_submission_information(options, app)
+
+      version.create_app_store_version_submission
 
       UI.success("Successfully submitted the app for review!")
     end
 
-    private def select_build(options)
-      app = options[:app]
-      app_version = options[:app_version]
-      v = app.edit_version(platform: options[:platform])
-
+    private def select_build(options, app, version, platform)
       if options[:build_number] && options[:build_number] != "latest"
         UI.message("Selecting existing build-number: #{options[:build_number]}")
-        if app_version
-          build = v.candidate_builds.detect { |a| a.build_version == options[:build_number] && a.train_version == app_version }
-        else
-          build = v.candidate_builds.detect { |a| a.build_version == options[:build_number] }
-        end
+
+        build = Spaceship::ConnectAPI::Build.all(
+          app_id: app.id,
+          version: options[:app_version],
+          build_number: options[:build_number],
+          platform: platform
+        ).first
+
         unless build
           UI.user_error!("Build number: #{options[:build_number]} does not exist")
         end
       else
         UI.message("Selecting the latest build...")
-        build = wait_for_build(app, app_version)
+        build = wait_for_build_processing_to_be_complete(app: app, platform: platform, options: options)
```

Comment:
@joshdholtz 

Thank you for your draft 👍
I tried this patch and found out this is not enough because old `wait_for_build` is more than `wait_for_build_processing_to_be_complete`.

`wait_for_build` get all build candidates and compare its upload_date to find latest_build before checking processing_to_be_complete.
When the fastlane user does not specify `app_version`, this changes will break their deliver.

Reply:
Ahhhh, thanks!! I will look into fixing that 🙂

Reply:
@nafu Candidate builds aren't really a thing anymore so I'm not sure how to do this 😔 A build doesn't appear on the API for a few minutes either. It takes some time to "pre-process" before it actually shows up as processing 😬 

I'm not sure what the correct solution is there. This _may_ have to be a breaking change to where an `app_version` is specified? Or if no `app_version` is specified maybe we show a list of builds the user can select from?

I never really used `deliver` this way so I'm not sure what would be the best approach. Your thoughts would be appreciated! 😊